extensions/amp-ad/0.1/amp-ad-batch-manager.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {user} from '../../../src/log';
+import {xhrFor} from '../../../src/xhr';
+
+/** @const {number} The number of milliseconds between checks to see if the response has arrived */
+const AD_BATCH_TICK_LENGTH = 50;
+
+/** @const {number} The number of ticks to wait for a response from the ad server before timing out */
+const AD_BATCH_RETRIES = 100;
+
+/** @const {!string} Tag name for ad batch manager implementation. */
+export const TAG_AD_BATCH_MANAGER = 'amp-ad-batch-manager';
```

Comment:
I don't understand, but done. It's only used for marking error messages: I mimicked what was in amp-ad-image.js. May I please point out that there is virtually no documentation of any of the parameters of the functions in log.js - I'm astonished that it could ever have passed a code review.